lib/OpenQA/UserAgent.pm

```diff
@@ -70,14 +70,18 @@ sub _add_auth_headers {
     my $timestamp = time;
     my %headers   = (
         Accept            => 'application/json',
-        'X-API-Microtime' => $timestamp
+        'X-API-Microtime' => $timestamp,
     );
     if ($self->apisecret && $self->apikey) {
         $headers{'X-API-Key'}  = $self->apikey;
         $headers{'X-API-Hash'} = hmac_sha1_sum($self->_path_query($tx) . $timestamp, $self->apisecret);
     }
 
     while (my ($k, $v) = each %headers) {
+        if (defined $tx->req->headers->header($k)) {
```

Comment:
I'd write this if as one-liner and wouldn't call `$tx->req->headers` now twice in each loop operation.

Reply:
ok, I canged it